webapp/components/PostCard/index.vue

```diff
@@ -1,69 +1,71 @@
 <template>
-  <ds-flex-item :width="width">
-    <ds-card
-      :image="post.image | proxyApiUrl"
-      :class="{ 'post-card': true, 'disabled-content': post.disabled }"
+  <ds-card
+    :image="post.image | proxyApiUrl"
+    :class="{ 'post-card': true, 'disabled-content': post.disabled }"
+  >
+    <!-- Post Link Target -->
+    <nuxt-link
+      class="post-link"
+      :to="{ name: 'post-id-slug', params: { id: post.id, slug: post.slug } }"
     >
-      <!-- Post Link Target -->
-      <nuxt-link
-        class="post-link"
-        :to="{ name: 'post-id-slug', params: { id: post.id, slug: post.slug } }"
-      >
-        {{ post.title }}
-      </nuxt-link>
-      <ds-space margin-bottom="small" />
-      <!-- Username, Image & Date of Post -->
-      <div>
-        <no-ssr>
-          <hc-user :user="post.author" :trunc="35" :date-time="post.createdAt" />
-        </no-ssr>
-        <hc-ribbon :text="$t('post.name')" />
+      {{ post.title }}
+    </nuxt-link>
+    <ds-space margin-bottom="small" />
+    <!-- Username, Image & Date of Post -->
+    <div>
+      <no-ssr>
+        <hc-user :user="post.author" :trunc="35" :date-time="post.createdAt" />
+      </no-ssr>
+      <hc-ribbon :text="$t('post.name')" />
+    </div>
+    <ds-space margin-bottom="small" />
+    <!-- Post Title -->
+    <ds-heading tag="h3" no-margin>{{ post.title }}</ds-heading>
+    <ds-space margin-bottom="small" />
+    <!-- Post Content Excerpt -->
+    <!-- eslint-disable vue/no-v-html -->
+    <!-- TODO: replace editor content with tiptap render view -->
+    <div class="hc-editor-content" v-html="excerpt" />
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-08-14T22:36:05Z" title="Thursday, August 15th 2019, 12:36:05 am +02:00">Aug 15, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -123,6 +132,25 @@ export default {
         this.$toast.error(err.message)
       }
     },
+    calculateItemHeight() {
+      const grid = document.querySelector('.masonry-grid')
```


We have to check if globals like `document` or `window` are available. Our tests are failing because of it.

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-08-14T22:38:57Z" title="Thursday, August 15th 2019, 12:38:57 am +02:00">Aug 15, 2019</time>_

---


Also I would find it cleaner in this case if the parent would interact with its children instead of the other way round. According to this diagram:
https://stackoverflow.com/questions/52433712/vue-js-component-lifecycle-for-children
![vue lifecycle hooks for parent components](https://i.stack.imgur.com/GjZ6a.png)
You could create a masonry component that iterates through all children with a particular class and sets the height. I don't know if this might work or if this still works if the list gets updated.. but worth a try.